### ISSUE TITLE
Confused by "else if" ?

### ISSUE THREAD

**Issue by rdebath** [koalaman/shellcheck#948]
#### For bugs
- Rule Id: SC1089
- My shellcheck version (`shellcheck --version` or "online"): version: 0.4.4
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

There are plenty of other warnings about this lovely piece of ... code.
But the `SC1089` seems to be a false positive.

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh

# shellcheck disable=SC2006,SC2039,SC2046,SC1117
if [ .`echo -n` = .-n ]
then
    echon() { echo "$1\c"; }
    echoe() { echo "$1\c"; }
else
    echon() { echo -n "$1"; }
    if [ .`echo -e` = .-e ]
    then echoe() { echo -n "$1"; }
    else if [ .`echo -e '\070\c'` = .8 ]
         then echoe() { echo -e "$1\c"; }
         else echoe() { echo -n -e "$1"; }
         fi
    fi
fi
if [ .`echoe '\070'` != .8 ]
then echoe(){ printf "%b" "$*" 2>/dev/null ; }
fi
if [ .`echoe '\171'` = .y ]
then o(){ echoe "`printf '\\\\%03o' $1`" ; }
else o(){ echoe "`printf '\\\\%04o' $1`" ; }
fi

```

#### Here's what shellcheck currently says:
```
Line 12:
    else if [ .`echo -e '\070\c'` = .8 ]
    ^-- SC1075: Use 'elif' instead of 'else if' (or put 'if' on new line if nesting).
 
Line 17:
fi
^-- SC1089: Parsing stopped here. Is this keyword correctly matched up?
```
#### Here's what I wanted or expected to see:
Only the SC1075 for this section, then it continues.

**Comment by koalaman** [comment]
You're right, of course. Your use of `else if` is perfectly correct, and ShellCheck is wrong.

Empirically, most of the time when people write `else if`, they think it's the same as `elif`. ShellCheck goes along with that -- even though it's incorrect -- because it gives these users better help. They clearly need it since they can't write `if` statements.

I know it's awful to ask anyone to change a perfectly correct script, but I hope you'll consider putting the `if` on a separate line as suggested to allow ShellCheck to be more helpful to people who don't know the syntax as well as you do.

**Comment by rdebath** [comment]
Well, usually I'd just disable `SC1075` and it would be done with. But if I do that I get a random syntax error from shellcheck for no visible reason. That hardly seems right or helpful.

**Comment by rdebath** [comment]
Actually, never mind, this looks about right; adding a pointless disable that actually doesn't disable the problem, except it does by just taking up space...

```sh
#!/bin/sh
:
# shellcheck disable=SC2006,SC2039,SC2046,SC1117
if [ .`echo -n` = .-n ]
then
    echon() { echo "$1\c"; }
    echoe() { echo "$1\c"; }
else
    echon() { echo -n "$1"; }
    if [ .`echo -e` = .-e ]
    then echoe() { echo -n "$1"; }
    else # shellcheck disable=SC1075 #
         if [ .`echo -e '\070\c'` = .8 ]
         then echoe() { echo -e "$1\c"; }
         else echoe() { echo -n -e "$1"; }
         fi
    fi
fi
if [ .`echoe '\070'` != .8 ]
then echoe(){ printf "%b" "$*" 2>/dev/null ; }
fi
if [ .`echoe '\171'` = .y ]
then o(){ echoe "`printf '\\\\%03o' $1`" ; }
else o(){ echoe "`printf '\\\\%04o' $1`" ; }
fi
```

**Comment by koalaman** [comment]
You don't need to disable anything if you insert a line break between the `else` and the `if`:

```
    if [ .`echo -e` = .-e ]
    then echoe() { echo -n "$1"; }
    else
         if [ .`echo -e '\070\c'` = .8 ]
         then echoe() { echo -e "$1\c"; }
         else echoe() { echo -n -e "$1"; }
         fi
    fi
```

**Comment by rdebath** [comment]
I know, that why I said it does nothing.
Except it does say why I'm using non-standard formatting.

**Comment by koalaman** [comment]
Ah. Yes, I agree that this is not ideal in any way.
